File: app/src/main/java/org/forgerock/auth/NoSessionRequestInterceptor.java
/*
 * Copyright (c) 2020 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package org.forgerock.auth;

import android.net.Uri;

import androidx.annotation.NonNull;

import org.forgerock.android.auth.Action;
import org.forgerock.android.auth.FRRequestInterceptor;
import org.forgerock.android.auth.Request;
import org.forgerock.android.auth.RequestInterceptor;

import static org.forgerock.android.auth.Action.AUTHENTICATE;

/**
 * Sample {@link RequestInterceptor} to add noSession
 */
public class NoSessionRequestInterceptor implements FRRequestInterceptor<Action> {

    @NonNull
    @Override
    public Request intercept(@NonNull Request request, Action tag) {
        if (tag.getType().equals(AUTHENTICATE)) {
            return request.newBuilder()
                    .url(Uri.parse(request.url().toString())
                            .buildUpon()
                            .appendQueryParameter("noSession", "true").toString())
                    .build();
        }
        return request;
    }
}

File: samples/app/src/main/java/com/example/app/setting/NoSessionRequestInterceptor.kt
/*
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 *  This software may be modified and distributed under the terms
 *  of the MIT license. See the LICENSE file for details.
 */
package com.example.app.setting

import android.net.Uri
import org.forgerock.android.auth.Action
import org.forgerock.android.auth.FRRequestInterceptor
import org.forgerock.android.auth.Request

/**
 * Sample [RequestInterceptor] to add noSession
 */
class NoSessionRequestInterceptor : FRRequestInterceptor<Action> {
    override fun intercept(request: Request, tag: Action): Request {
        return if (tag.type == Action.AUTHENTICATE) {
            request.newBuilder()
                .url(Uri.parse(request.url().toString())
                    .buildUpon()
                    .appendQueryParameter("noSession", "true").toString())
                .build()
        } else request
    }
}
